ownCloud/Client/Actions/Actions+Extensions/CreateDocumentAction.swift

```diff
@@ -0,0 +1,203 @@
+			completed(with: NSError(ocError: .internal))
+			return
+		}
+
+		guard let documentTypes = context.core?.appProvider?.types?.filter({ fileType in
+			return fileType.allowCreation
+		}).sorted(by: { (type1, type2) in
+			if let name1 = type1.name, let name2 = type2.name {
+				return name1.localizedCaseInsensitiveCompare(name2) == .orderedAscending
+			}
+			return false
+		}), documentTypes.count > 0 else {
+			completed()
+			return
+		}
+
+		OnMainThread {
+			let documentTypesDataSource = OCDataSourceArray()
+			let documentTypesSection = CollectionViewSection(identifier: "documentTypes", dataSource: documentTypesDataSource, cellStyle: .init(with: .fillSpace), cellLayout: .fullWidth(itemHeightDimension: .estimated(54), groupHeightDimension: .estimated(54), edgeSpacing: NSCollectionLayoutEdgeSpacing(leading: .fixed(0), top: .fixed(10), trailing: .fixed(0), bottom: .fixed(10)), contentInsets: NSDirectionalEdgeInsets(top: 10, leading: 20, bottom: 10, trailing: 20)), clientContext: self.context.clientContext)
+			let documentPickerViewController = CollectionViewController(context: self.context.clientContext, sections: [ documentTypesSection ])
+
+			let navigationViewController = ThemeNavigationController(rootViewController: documentPickerViewController)
+			navigationViewController.modalPresentationStyle = .formSheet
+			viewController.present(navigationViewController, animated: true)
+
+			documentPickerViewController.title = "New document".localized
+			documentPickerViewController.navigationItem.rightBarButtonItem = UIBarButtonItem(systemItem: .cancel, primaryAction: UIAction(handler: { [weak navigationViewController] action in
+				navigationViewController?.dismiss(animated: true)
+				self.completed()
+			}), menu: nil)
+
+			let createDocument : (OCAppProviderFileType) -> Void = { (fileType) in
+				guard let core = self.core, let parentItem = try? core.cachedItem(at: itemLocation) else {
+					self.completed()
+					return
+				}
+
+				core.suggestUnusedNameBased(on: "New document".localized.appending((fileType.extension != nil) ? ".\(fileType.extension!)" : ""), at: itemLocation, isDirectory: false, using: .numbered, filteredBy: nil, resultHandler: { (suggestedName, _) in
+					guard let suggestedName = suggestedName else { return }
+
+					OnMainThread {
+						let documentNameViewController = NamingViewController( with: self.core, defaultName: suggestedName, stringValidator: { name in
+							if name.contains("/") || name.contains("\\") {
+								return (false, nil, "File name cannot contain / or \\".localized)
+							} else {
+								if let item = item {
+									if ((try? self.core?.cachedItem(inParent: item, withName: name, isDirectory: true)) != nil) ||
+									   ((try? self.core?.cachedItem(inParent: item, withName: name, isDirectory: false)) != nil) {
+										return (false, "Item with same name already exists".localized, "An item with the same name already exists in this location.".localized)
```

Comment:
Localisation is missing in strings file

Reply:
This was added in `feature/new-navigation`.